### PR TITLE
Set aria-hidden on outside elements for popovers and fix focusing dialogs in iOS VoiceOver

### PR DESCRIPTION
**Pull request by devongovett**
This fixes several issues with popovers and overlay triggers in iOS VoiceOver:

* We use portals at the end of the document body for all overlays, but only set `aria-hidden` on the surrounding content for modals and trays currently. For popovers/menus, this led to a confusing experience where you could use the screen reader cursor to navigate outside the popover/menu without closing it. Since the portal is at the end of the document, this was confusing because you'd navigate to things at the end of the document, far beyond the original trigger that opened it. Now we set `aria-hidden` on surrounding content for all overlay types, including popovers. In addition, this now means all dialogs are modal (have `aria-modal`), so we can simplify our code a bit.

* Dialogs were not receiving VoiceOver cursor focus or announcing that they had opened on iOS. The dialog was actually keyboard focused properly, but VoiceOver would not move its cursor or announce because of the animation. We now re-focus the dialog after a delay to jolt VoiceOver into doing the right thing.

* Focus was not being restored back to the trigger properly because Safari would not focus the trigger with VoiceOver enabled. `usePress` now accounts for this in its `onClick` handler.

### CODE REVIEW

File: packages/@react-aria/overlays/test/usePreventScroll.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    expect(document.body).not.toHaveStyle('overflow: hidden');
+
+    let res = render(<Example />);
+    expect(document.body).toHaveStyle('overflow: hidden');
+
+    res.unmount();
+    expect(document.body).not.toHaveStyle('overflow: hidden');
+  });
+
+  it('should work with nested modals', function () {
```
Comment (snowystinger) [inline packages/@react-aria/overlays/test/usePreventScroll.test.js]: are these nested? or just adjacent? I realize that a nested modal trigger would result in two modals as siblings but do we want to call the test nested for that?

File: packages/@react-spectrum/dialog/src/Dialog.tsx [hunk 2]
```diff
@@ -38,60 +46,19 @@ export function Dialog(props: SpectrumDialogProps) {
...
-
-let sizeMap = {
-  S: 'small',
-  M: 'medium',
-  L: 'large',
-  fullscreen: 'fullscreen',
-  fullscreenTakeover: 'fullscreenTakeover'
-};
-
-function BaseDialog({children, slots, size, role, ...otherProps}: SpectrumBaseDialogProps) {
```
Comment (snowystinger) [inline packages/@react-spectrum/dialog/src/Dialog.tsx]: yay!

File: packages/@react-aria/dialog/src/useDialog.ts [hunk 3]
```diff
@@ -32,11 +32,26 @@ export function useDialog(props: DialogProps): DialogAria {
   useEffect(() => {
     if (ref.current && !ref.current.contains(document.activeElement)) {
       focusWithoutScrolling(ref.current);
+
+      // Safari on iOS does not move the VoiceOver cursor to the dialog
+      // or announce that it has opened until it has rendered. A workaround
+      // is to wait for half a second, then blur and re-focus the dialog.
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-aria/dialog/src/useDialog.ts]: nope. in my testing, anything under 500ms does not work.

File: packages/@react-aria/interactions/src/usePress.ts [hunk 4]
```diff
@@ -206,6 +206,11 @@ export function usePress(props: PressHookProps): PressResult {
           // If triggered from a screen reader or by using element.click(),
           // trigger as if it were a keyboard click.
           if (!state.ignoreClickAfterPress && !state.ignoreEmulatedMouseEvents && isVirtualClick(e.nativeEvent)) {
+            // Ensure the element receives focus (VoiceOver on iOS does not do this)
+            if (!isDisabled) {
+              focusWithoutScrolling(e.currentTarget);
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-aria/interactions/src/usePress.ts]: last point in the PR description. 😉

File: packages/@react-aria/overlays/test/usePreventScroll.test.js (hunk 1 above)
Comment (devongovett) [reply packages/@react-aria/overlays/test/usePreventScroll.test.js]: hmm I just renamed this file, this is not a new test :)

**Comment by dannify** [comment]
Pickup your game IOS voiceover 👿

**Pull request merged**